plugin/inlay_hint.py

```diff
@@ -142,12 +143,15 @@ def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uui
             })
             result += f'<a href="{inlay_hint_click_command}">'
         instruction_text = '\nDouble-click to insert' if has_text_edits else ""
-        result += f'<span title="{(tooltip + instruction_text).strip()}">{html.escape(label)}</span>'
+        truncated_label = label[:truncate_limit] + '...' if len(label) >= truncate_limit else label
+        result += f'<span title="{(tooltip + instruction_text).strip()}">{html.escape(truncated_label)}</span>'
         if is_clickable:
             result += "</a>"
         return result
 
     for label_part in label:
+        if truncate_limit <= 0:
+            break
```

Comment:
kinda weird to completely skip this part of the code if limit is out of range. I would expect falling back to default value in this case.

Reply:
We iterate over `label` which contains "parts". And we need to "exit" the loop if we cannot cut any more. 

For example: `: { session: { user: AdapterUser } }` will be split into parts like:
- `:`
- `{`
- ` `
- `session`
- `:`
- ` `
- `{`
...

We need to decrement `truncate_limit` each iteration and stop iterating when we reach the end of the truncation.

I think decrementing the method parameter can lead to confusion, so I will move truncate_limit to a local variable to make it more intuitive.

Reply:
Done (probably) in: [6335b74](https://github.com/sublimelsp/LSP/pull/2514/commits/6335b74522c5de8803f2c0349a45fe62392fd5b0)

I hope it will make sense now, please let me know if you have better approach